runtime/compiler/control/CompilationController.cpp

```diff
@@ -203,25 +203,21 @@ TR_OptimizationPlan *TR::DefaultCompilationStrategy::processEvent(TR_MethodEvent
          // For sync re-compilation we have attached a plan to the persistentBodyInfo
          bodyInfo = TR::Recompilation::getJittedBodyInfoFromPC(event->_oldStartPC);
          methodInfo = bodyInfo->getMethodInfo();
-         if (bodyInfo->getUsesJProfiling() && !bodyInfo->getIsProfilingBody())
+         if (methodInfo->getReasonForRecompilation() == TR_PersistentMethodInfo::RecompDueToInlinedMethodRedefinition)
+            methodInfo->incrementNumberOfInlinedMethodRedefinition();
```

Comment:
Is it possible for several threads to enter this code more or less concurrently and therefore increment the counter even though we are actually dealing with the same recompilation event? If that is the case a better solution would be to increment this counter at the beginning of the compilation, right before the code that tests whether we need to disable nextgenHCR